src/Gemstone.Data/Model/TableOperations.cs

```diff
@@ -1180,16 +1577,40 @@ s_encryptDataTargets is not null &&
     /// <param name="page">Desired page of records.</param>
     /// <param name="pageSize">Desired page size.</param>
     /// <returns>A page of records.</returns>
-    public IEnumerable<T> GetPageOfRecords(T[] records, int page, int pageSize)
+    public IEnumerable<T?> GetPageOfRecords(T?[] records, int page, int pageSize)
     {
         return records.ToPagedList(page, pageSize, records.Length);
     }
 
-    IEnumerable ITableOperations.GetPageOfRecords(object[] records, int page, int pageSize)
+    IEnumerable ITableOperations.GetPageOfRecords(object?[] records, int page, int pageSize)
+    {
+        try
+        {
+            return GetPageOfRecords(records.Cast<T?>().ToArray(), page, pageSize);
+        }
+        catch (InvalidCastException ex)
+        {
+            throw new ArgumentException($"One of the provided records cannot be converted to type \"{typeof(T).Name}\": {ex.Message}", nameof(records), ex);
+        }
+    }
+
+    /// <summary>
+    /// Gets the specified <paramref name="page"/> of records from the provided source <paramref name="records"/> array.
+    /// </summary>
+    /// <param name="records">Source records array.</param>
+    /// <param name="page">Desired page of records.</param>
+    /// <param name="pageSize">Desired page size.</param>
+    /// <returns>A page of records.</returns>
+    public IAsyncEnumerable<T?> GetPageOfRecordsAsync(IAsyncEnumerable<T?> records, int page, int pageSize)
+    {
+        return records.Skip(page * pageSize).Take(pageSize);
```

Comment:
This is probably really slow. Though I'm not sure what to do about that.

Reply:
It's not referenced by library code, but provided in case end-user needs to manually paginate through a set of records. The async version was simply added for parity with `GetPageOfRecords`. I am not overly concerned with its speed given use may be rare.